src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -979,12 +981,12 @@ void MeshForwarder::UpdateRoutes(uint8_t *           aFrame,
 
     VerifyOrExit(netif.GetLowpan().DecompressBaseHeader(ip6Header, aMeshSource, aMeshDest, aFrame, aFrameLength) > 0);
 
-    netif.GetAddressResolver().UpdateCacheEntry(ip6Header.GetSource(), meshHeader.GetSource());
+    netif.GetAddressResolver().UpdateCacheEntry(ip6Header.GetSource(), aMeshSource.GetShort());
 
     neighbor = netif.GetMle().GetNeighbor(ip6Header.GetSource());
     VerifyOrExit(neighbor != NULL && !neighbor->IsFullThreadDevice());
 
-    if (Mle::Mle::GetRouterId(meshHeader.GetSource()) != Mle::Mle::GetRouterId(GetNetif().GetMac().GetShortAddress()))
+    if (Mle::Mle::GetRouterId(aMeshSource.GetShort()) != Mle::Mle::GetRouterId(GetNetif().GetMac().GetShortAddress()))
```

Comment:
This appears to read that if the Router ID corresponding to the Source RLOC16 in the Mesh header doesn't match our own, then we remove the neighbor corresponding to that Source address. Why doesn't this remove all neighbors?

Reply:
Remove all neighbors?  Can you elaborate?

Reply:
@smides-nest , we reach this check only if the src IPv6 address corresponds to a neighbor which is a child (which would be then our child). Then we check if its RLOC matches ours, if not then it has switched to a new parent and is removed.

Reply:
Got it. I wasn't aware that the Source address here was *from* the child. Thanks.